Calculator/Calculator/Controller/CalculatorViewController.swift

```diff
@@ -14,149 +14,175 @@ class CalculatorViewController: UIViewController {
-        }
+        currentOperator = operatorSymbol
+        currentOperand = CalculatorValue.emptyArray
         
-        operandNumber = CalculatorValue.emptyArray
-        formulaNumber = CalculatorValue.emptyArray
+        resetUI()
     }
     
-    private func addListStackView() -> UIStackView {
-        let operatorLabel = UILabel()
-        let operandLabel = UILabel()
-        let stackView = UIStackView()
+    @IBAction private func allClearButtonDidTap(_ sender: Any) {
+        inputString = CalculatorValue.emptyArray
+        currentOperator = CalculatorValue.emptyArray
+        currentOperand = CalculatorValue.emptyArray
+        resultString = CalculatorValue.emptyArray
         
-        stackView.spacing = 8
-        operandLabel.textColor = .white
-        operatorLabel.textColor = .white
-        operandLabel.font = UIFont.preferredFont(forTextStyle: .title3)
-        operatorLabel.font = UIFont.preferredFont(forTextStyle: .title3)
-        operatorLabel.text = self.operatorLabel.text
-        operandLabel.text = self.operandLabel.text
-        stackView.addArrangedSubview(operatorLabel)
-        stackView.addArrangedSubview(operandLabel)
+        resetUI()
         
-        return stackView
+        scrollStackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
     }
     
-    private func addScrollStackView() {
-        scrollStackView.addArrangedSubview(addListStackView())
-        scrollToBottom()
+    @IBAction private func ClearEntryButtonDidTap(_ sender: Any) {
+        currentOperand = CalculatorValue.emptyArray
+        resetUI()
     }
     
-    private func scrollToBottom() {
-        formulaScrollView.layoutIfNeeded()
-        formulaScrollView.setContentOffset(CGPoint(x: 0, y: formulaScrollView.contentSize.height - formulaScrollView.bounds.height), animated: false)
+    @IBAction private func plusMinusButtonDidTap(_ sender: Any) {
+        if currentOperand.first == Character(CalculatorValue.minus) {
+            currentOperand.remove(at: currentOperand.startIndex)
+        } else {
+            currentOperand = CalculatorValue.minus + currentOperand
+        }
+        
+        if currentOperand == CalculatorValue.minus {
+            operandLabel.text = currentOperand + CalculatorValue.zero
+        } else {
+            operandLabel.text = currentOperand
+        }
     }
     
-    private func numberFormatter(operand: String) -> String {
-        guard let doubleOperand = Double(operand) else { return CalculatorValue.emptyArray }
-        let numberFormatter = NumberFormatter()
-        numberFormatter.numberStyle = .decimal
-        numberFormatter.maximumFractionDigits = 20
+    @IBAction private func equalButtonDidTap(_ sender: Any) {
+        inputString += currentOperator + currentOperand
         
-        return numberFormatter.string(from: NSNumber(value: doubleOperand)) ?? CalculatorValue.emptyArray
+        guard !inputString.contains(CalculatorValue.nan) else {
```

Comment:
개인적으로 논리부정연산자(!)는 가독성이 좋지 않다고 생각합니다.
취향이니 참고만 해주세요.

```suggestion
        guard inputString.contains(CalculatorValue.nan) == false else {
```